### PR TITLE
clang-tidy cleanups

### PR DESCRIPTION
**Pull request by neheb**
Suggested by useStlAlgorithm

Signed-off-by: Rosen Penev <rosenp@gmail.com>

I'm not sure these are correct. Would love someone to check these.

### CODE REVIEW

**Comment by neheb** [comment]
Removed algorithm commit and added others. That commit was totally broken anyway...

File: src/util/upnp_clients.cc [hunk 1]
```diff
@@ -156,7 +156,7 @@ void Clients::getInfo(const struct sockaddr_storage* addr, const std::string& us
     // by addClientByDiscovery
     if (!found) {
         // always return something, 'Unknown' if we do not know better
-        assert(clientInfo[0].type == ClientType::Unknown);
+        static_assert(clientInfo[0].type == ClientType::Unknown);
```
Comment (whyman) [inline src/util/upnp_clients.cc:159]: I think we want to assert at runtime still

Comment (neheb) [reply src/util/upnp_clients.cc:159]: Why?

**Comment by neheb** [comment]
Rebased.

File: src/util/upnp_clients.cc (hunk 1 above)
Comment (whyman) [reply src/util/upnp_clients.cc:159]: Because the check is a runtime one and makes no sense at compile time?

Comment (neheb) [reply src/util/upnp_clients.cc:159]: The way I read this is that it asserts that the first struct member at 0 is Unknown, which it is. Either that or the check is wrong.

Comment (whyman) [reply src/util/upnp_clients.cc:159]: Yes, but why would we want to check that at compile time?

Comment (neheb) [reply src/util/upnp_clients.cc:159]: It’s a const struct...

**Pull request merged**